src/test/java/bio/terra/integration/AzureIntegrationTest.java

```diff
@@ -384,23 +378,47 @@ private void populateOmopTable() throws Exception {
   public void testSnapshotBuilder() throws Exception {
     populateOmopTable();
 
-    // Test getConcepts
-    var getConceptResponse = dataRepoFixtures.getConcepts(steward, datasetId, 2);
-    var concepts = getConceptResponse.getResult();
+    var concept1 =
+        new SnapshotBuilderConcept().name("concept1").id(1).count(22).code("11").hasChildren(false);
+    var concept3 =
+        new SnapshotBuilderConcept().name("concept3").id(3).count(24).code("13").hasChildren(true);
+
+    getConceptChildrenTest(concept1, concept3);
+    searchConceptTest(concept1);
+    getConceptHierarchyTest(concept1, concept3);
+  }
 
-    var concept1 = concepts.get(0);
-    var concept3 = concepts.get(1);
+  private void searchConceptTest(SnapshotBuilderConcept concept1) throws Exception {
+    var searchConceptsResult =
+        dataRepoFixtures.searchConcepts(steward, datasetId, 19, concept1.getName());
+    // A concept returned by search concepts always has hasChildren = true, even if it doesn't
+    // have children.
+    var concept1WithChildren =
```

Comment:
Do you mean should I overwrite the parameter `concept1` instead of declaring a new variable name?

This test can't overwrite `hasChildren` on `concept1` since other tests may need the real value for this concept, which is `false`.